kevm-pyk/src/tests/unit/test_foundry_list.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+import json
+from functools import cached_property
+from os import listdir
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from kevm_pyk.foundry import foundry_list
+from kevm_pyk.solc_to_k import Contract
+from tests.unit.utils import TEST_DATA_DIR
+
+# from pyk.proof.reachability import APRProof
+
+
+# from unittest.mock import patch
+
+
+if TYPE_CHECKING:
+    from typing import Final
+
+    from pytest_mock import MockerFixture
+
+
+LIST_DATA_DIR: Final = TEST_DATA_DIR / 'foundry-list'
+LIST_OUT: Final = LIST_DATA_DIR / 'out/'
+LIST_APR_PROOF: Final = LIST_OUT / 'apr_proofs/'
+LIST_EXPECTED: Final = LIST_DATA_DIR / 'foundry-list.expected'
+
+
+class FoundryMock:
+    out = LIST_OUT
+
+    @cached_property
+    def contracts(self) -> dict[str, Contract]:
+        ret: dict[str, Contract] = {}
+        for proof_file in listdir(LIST_APR_PROOF):
+            proof_path = LIST_APR_PROOF / proof_file
+            proof_dict = json.loads(Path(proof_path).read_text())
+            pid = proof_dict['id']
+            contract = Contract.__new__(Contract)
+            full_method = pid.split(':')[0]
+            method = Contract.Method.__new__(Contract.Method)
+            contract.name, method.name = full_method.split('.')
+            if not hasattr(contract, 'methods'):
+                contract.methods = ()
+            contract.methods = contract.methods + (method,)
+            ret[proof_file] = contract
+        return ret
+
+    def proof_digest(self, contract: str, test: str) -> str:
+        for proof_file in listdir(LIST_APR_PROOF):
+            proof_path = LIST_APR_PROOF / proof_file
+            proof_dict = json.loads(Path(proof_path).read_text())
+            pid = proof_dict['id']
+            if pid.startswith(f'{contract}.{test}'):
+                return pid
+        return ''
+
+
+def test_foundry_list(mocker: MockerFixture, update_expected_output: bool) -> None:
+    foundry_mock = mocker.patch('kevm_pyk.foundry.Foundry')
+    foundry_mock.return_value = FoundryMock()
```

Comment:
I think this should be

```suggestion
    foundry_mock.return_value = FoundryMock
```

Reply:
Well then considering the other comment it should now be `FoundryMock(LIST_OUT)`, right ?

Reply:
Well, we're mocking a class with another, so I'd expect it to work with just `FoundryMock`. Can you check to make sure?

Reply:
Oh I get it, I think that in the above review you meant

```python
class FoundryMock:
   out: Path

    def __init__(self)
        sef.out = LIST_OUT
```

Reply:
Calling `FoundryMock` does not seem to invoke the `__init__`

Reply:
Oh yeah, that makes sense. Maybe we should use `monkeypatch` then, that's what we're using `mocker` for anyways:

```python3
from pytest import MonkeyPatch

import foundry

def test(monkeypatch: MonkeyPatch):
    monkeypatch.setattr(foundry, 'Foundry', FoundryMock)
```

But the current approach is fine as well.

Reply:
Interesting, please tell me if the current implementation fits